core/frontend/src/ModelState.ts

```diff
@@ -170,18 +174,130 @@ export abstract class GeometricModelState extends ModelState implements Geometri
 
     if(tilesetUrl) {
       const rdSourceKeyCS = RealityDataSource.createKeyFromUrl(tilesetUrl);
-      return createRealityTileTreeReference({
+      tileTreeReference = createRealityTileTreeReference({
         rdSourceKey: rdSourceKeyCS,
         url : tilesetUrl,
         iModel: this.iModel,
         source: view,
         modelId: this.id,
         tilesetToDbTransform: this.jsonProperties.tilesetToDbTransform,
         classifiers: undefined !== spatialModel ? spatialModel.classifiers : undefined,
+        realityDataDisplayStyle: undefined !== spatialModel ? spatialModel.realityDataDisplayStyle : undefined,
       });
     }
 
-    return createPrimaryTileTreeReference(view, this);
+    if (tileTreeReference === undefined) {
+      return createPrimaryTileTreeReference(view, this);
+    }
+
+    // Try to query clips from the iModel
+    if (undefined !== this.id && Id64.isValidId64(this.id) && !Id64.isTransient(this.id)) {
+      const clipEcsql = `SELECT SmModelClips FROM ScalableMesh.ScalableMeshModel WHERE ECInstanceId=${this.id}`;

```

Comment:
Why are you querying twice instead of using a single query to obtain both properties?

Reply:
It was easier for me to separate the 2 calls, one for each different clip definitions but I realize now that it's more efficient to make a single call. I will change the code so to have just one call.

Reply:
Done, see 06c9c7d.